ledger/participant-integration-api/src/main/scala/platform/store/backend/common/MeteringStorageBackendTemplate.scala

```diff
@@ -70,15 +70,17 @@ private[backend] object MeteringStorageBackendTemplate {
 
   }
 
-  /** Oracle does not understand true/false so compare against number
-    * @return 0 if the option is unset and non-zero otherwise
+  /**  Evaluate to the passed condition if the option is non-empty or return true otherwise
     */
-  def isSet(o: Option[_]): Int = o.fold(0)(_ => 1)
+  def ifSet[A](o: Option[A], expr: A => CompositeSql): CompositeSql = {
+    o.fold(cSQL"1=1")(expr)
```

Comment:
with the composite approach you could extend the scope to the `and` as well, and not generate a `AND true` in vain

Reply:
I did consider this approach and saw it used in other places.  What I liked about the way I did it was that:
- it makes the SQL read more cleanly
- the 'AND' based approach does not work, in isolation, in all situations (e.g. when the ifSet is the only codition).